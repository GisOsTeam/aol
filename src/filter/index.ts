/**
 * @file Automatically generated by barrelsby.
 */

export * from './FilterBuilder';
export * from './IFilter';
