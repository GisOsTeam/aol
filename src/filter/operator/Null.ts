import { IOperator, OperatorEnum, OperatorType } from './IOperator';

export class Null implements IOperator {
  public not: boolean;
  public readonly type: OperatorType = OperatorEnum.null;

  constructor(not = false) {
    this.not = not;
  }

  public toString(): string {
    if (this.not) {
      return 'IS NOT NULL';
    }
    return 'IS NULL';
  }
}
