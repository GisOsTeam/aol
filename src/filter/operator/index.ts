/**
 * @file Automatically generated by barrelsby.
 */

export * from './AndOp';
export * from './Equal';
export * from './GreaterOrEqualThan';
export * from './GreaterThan';
export * from './IOperator';
export * from './Ilike';
export * from './In';
export * from './Like';
export * from './LowerOrEqualThan';
export * from './LowerThan';
export * from './Null';
export * from './Or';
