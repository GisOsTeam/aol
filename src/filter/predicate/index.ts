/**
 * @file Automatically generated by barrelsby.
 */

export * from './AndPre';
export * from './BasePredicate';
export * from './Equal';
export * from './FilterPredicate';
export * from './GreaterOrEqualThan';
export * from './GreaterThan';
export * from './IPredicate';
export * from './Ilike';
export * from './In';
export * from './Like';
export * from './LowerOrEqualThan';
export * from './LowerThan';
export * from './Or';
