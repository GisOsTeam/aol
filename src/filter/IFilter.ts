import { IOperator } from './operator';

export enum FieldTypeEnum {
  Oid = 'Oid',
  Boolean = 'Boolean',
  Number = 'Number',
  String = 'String',
  Date = 'Date',
  Geometry = 'Geometry',
  Unknown = 'Unknown',
}

export type FieldType =
  | FieldTypeEnum.Oid
  | FieldTypeEnum.Boolean
  | FieldTypeEnum.Number
  | FieldTypeEnum.String
  | FieldTypeEnum.Date
  | FieldTypeEnum.Geometry
  | FieldTypeEnum.Unknown;

export interface IField<T> {
  key: string | keyof T;
  type: FieldType;
  alias?: string;
}

export interface IFilter<T> {
  field: IField<T>;
  operator: IOperator;
  value: string | number | boolean | any[];
}

export interface IPredicate<T> extends IFilter<T> {
  toString(filterBuilderType?: FilterBuilderType): string;
}

export enum FilterBuilderTypeEnum {
  CQL = 'CQL',
  SQL = 'SQL',
  OGC = 'OGC',
}

export type FilterBuilderType = FilterBuilderTypeEnum.CQL | FilterBuilderTypeEnum.SQL | FilterBuilderTypeEnum.OGC;
