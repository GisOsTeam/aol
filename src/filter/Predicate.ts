import { FieldTypeEnum, FilterBuilderType, FilterBuilderTypeEnum, IField, IFilter, IPredicate } from './IFilter';
import { IOperator, OperatorEnum } from './operator';

export class Predicate<T> implements IPredicate<T> {
  public field: IField<T>;
  public operator: IOperator;
  public value: string | number | boolean | any[];

  constructor(filter: IFilter<T>) {
    this.field = filter.field;
    this.operator = filter.operator;
    this.value = filter.value;
  }

  public toString(filterBuilderType?: FilterBuilderType): string {
    if (
      (typeof this.value === 'number' || typeof this.value === 'boolean') &&
      (this.field.type === FieldTypeEnum.Number || this.field.type === FieldTypeEnum.Boolean) &&
      this.operator.type === OperatorEnum.equal
    ) {
      return `(${this.field.key} ${this.operator.toString(filterBuilderType)} ${this.getValue(filterBuilderType)})`;
    } else {
      if (this.operator.type === OperatorEnum.in) {
        return `(${this.field.key} ${this.operator.toString(filterBuilderType)} (${this.getValue(filterBuilderType)}))`;
      } else if (filterBuilderType === FilterBuilderTypeEnum.CQL) {
        this.field.key = `Concatenate(${this.field.key})`;
      } else if (this.operator.type === OperatorEnum.ilike) {
        if (filterBuilderType === FilterBuilderTypeEnum.SQL) {
          this.field.key = `UPPER(${this.field.key})`;
        }
      }
      return `(${this.field.key} ${this.operator.toString(filterBuilderType)} '${this.getValue(filterBuilderType)}')`;
    }
  }

  private getValue(filterBuilderType?: FilterBuilderType): string | number | boolean | any[] {
    if (this.operator.type === OperatorEnum.in) {
      return (this.value as any[]).map((v) => (typeof v === 'string' ? `'${v}'` : v)).join(',');
    } else if (this.operator.type === OperatorEnum.ilike) {
      if (filterBuilderType === FilterBuilderTypeEnum.SQL) {
        if (typeof this.value === 'number') {
          return `${this.value}`.toLowerCase();
        } else if (typeof this.value === 'string') {
          return this.value.toUpperCase();
        } else {
          console.error('unsupported value type for ilike operator');
        }
      }
    }
    return this.value;
  }
}
