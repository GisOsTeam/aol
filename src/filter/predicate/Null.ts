import { FilterBuilderType, IField } from '../IFilter';
import { Null as NullOp } from '../operator';
import { FilterPredicate } from './FilterPredicate';

export class Null<T> extends FilterPredicate<T, NullOp> {
  constructor(leftHand: IField<T>, operator: NullOp) {
    super(leftHand, operator, undefined);
  }

  public toString(type?: FilterBuilderType): string {
    return `(${this.buildLeftHandString(type)} ${this.operator.toString()})`;
  }

  protected buildLeftHandString(type: FilterBuilderType): string {
    return this.defaultLeftHandString(type);
  }

  protected buildRightHandString(type: FilterBuilderType): string {
    return this.defaultRightHandString();
  }
}
