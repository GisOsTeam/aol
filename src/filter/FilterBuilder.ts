import { FilterBuilderType } from './IFilter';
import { AndPre, IPredicate, Or } from './predicate';

export class FilterBuilder {
  public predicate: IPredicate;

  constructor(predicate: IPredicate) {
    this.predicate = predicate;
  }

  public static build(predicate: IPredicate, type: FilterBuilderType): string {
    if (!predicate) {
      return 'INCLUDE';
    }
    return predicate.toString(type);
  }

  public build(type: FilterBuilderType): string {
    if (!this.predicate) {
      return 'INCLUDE';
    }
    return this.predicate.toString(type);
  }

  public and<RP extends IPredicate>(rightPredicate: RP): FilterBuilder {
    return new FilterBuilder(new AndPre(this.predicate, rightPredicate));
  }

  public or<RP extends IPredicate>(rightPredicate: RP): FilterBuilder {
    return new FilterBuilder(new Or<any, RP>(this.predicate, rightPredicate));
  }
}
