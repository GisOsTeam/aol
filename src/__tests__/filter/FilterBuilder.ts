import {
  AndOp,
  Equal as EqualOp,
  GreaterOrEqualThan as GreaterOrEqualThanOp,
  GreaterThan as GreaterThanOp,
  Ilike as IlikeOp,
  In as InOp,
  Like as LikeOp,
  LowerOrEqualThan as LowerOrEqualThanOp,
  LowerThan as LowerThanOp,
  Null as NullOp,
  Or as OrOp,
} from '../../filter/operator';
import { FieldTypeEnum, FilterBuilder, FilterBuilderTypeEnum, IField } from '../../filter';
import {
  AndPre,
  Equal,
  GreaterOrEqualThan,
  GreaterThan,
  Ilike,
  In,
  Like,
  LowerOrEqualThan,
  LowerThan,
  Or,
} from '../../filter/predicate';
import { Null } from '../../filter/predicate/Null';

const numberField: IField<{ foo: number }> = {
  type: FieldTypeEnum.Number,
  key: 'foo',
};
const stringField: IField<{ foo: string }> = {
  type: FieldTypeEnum.String,
  key: 'foo',
};
const booleanField: IField<{ foo: boolean }> = {
  type: FieldTypeEnum.Boolean,
  key: 'foo',
};

describe('aol.filter', () => {
  describe('operator', () => {
    test('And', () => {
      let operator = new AndOp();
      expect<string>(operator.toString()).toEqual('AND');
      operator = new AndOp(true);
      expect<string>(operator.toString()).toEqual('AND NOT');
    });
    test('Equal', () => {
      let operator = new EqualOp();
      expect<string>(operator.toString()).toEqual('=');
      operator = new EqualOp(true);
      expect<string>(operator.toString()).toEqual('<>');
    });
    test('GreaterThan', () => {
      const operator = new GreaterThanOp();
      expect<string>(operator.toString()).toEqual('>');
    });
    test('GreaterOrEqualThan', () => {
      const operator = new GreaterOrEqualThanOp();
      expect<string>(operator.toString()).toEqual('>=');
    });
    test('Ilike', () => {
      let operator = new IlikeOp();
      expect<string>(operator.toString(FilterBuilderTypeEnum.SQL)).toEqual('LIKE');
      expect<string>(operator.toString()).toEqual('ILIKE');
      operator = new IlikeOp(true);
      expect<string>(operator.toString(FilterBuilderTypeEnum.SQL)).toEqual('NOT LIKE');
      expect<string>(operator.toString()).toEqual('NOT ILIKE');
    });
    test('In', () => {
      let operator = new InOp();
      expect<string>(operator.toString()).toEqual('IN');
      operator = new InOp(true);
      expect<string>(operator.toString()).toEqual('NOT IN');
    });
    test('Like', () => {
      let operator = new LikeOp();
      expect<string>(operator.toString()).toEqual('LIKE');
      operator = new LikeOp(true);
      expect<string>(operator.toString()).toEqual('NOT LIKE');
    });
    test('LowerThan', () => {
      const operator = new LowerThanOp();
      expect<string>(operator.toString()).toEqual('<');
    });
    test('LowerOrEqualThan', () => {
      const operator = new LowerOrEqualThanOp();
      expect<string>(operator.toString()).toEqual('<=');
    });
    test('Null', () => {
      let operator = new NullOp();
      expect<string>(operator.toString()).toEqual('IS NULL');
      operator = new NullOp(true);
      expect<string>(operator.toString()).toEqual('IS NOT NULL');
    });
    test('Or', () => {
      let operator = new OrOp();
      expect<string>(operator.toString()).toEqual('OR');
      operator = new OrOp(true);
      expect<string>(operator.toString()).toEqual('OR NOT');
    });
  });

  describe('predicate', () => {
    describe('and', () => {
      describe('cql', () => {
        // TODO
      });
      describe('sql', () => {
        // TODO
      });
    });

    describe('equal', () => {
      describe('cql', () => {
        test('not.number', () => {
          const predicate = new Equal(numberField, new EqualOp(true), 1);
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });

        test('number', () => {
          const predicate = new Equal(numberField, new EqualOp(), 1);
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });

        test('not.string', () => {
          const predicate = new Equal(stringField, new EqualOp(true), '1');
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });

        test('string', () => {
          const predicate = new Equal(stringField, new EqualOp(), '1');
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });

        test('not.boolean', () => {
          const predicate = new Equal(booleanField, new EqualOp(true), true);
          expect(FilterBuilder.build(predicate, FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });

        test('boolean', () => {
          const predicate = new Equal(booleanField, new EqualOp(), true);
          expect(FilterBuilder.build(predicate, FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });
      });
      describe('sql', () => {
        test('not.number', () => {
          const predicate = new Equal(numberField, new EqualOp(true), 1);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('number', () => {
          const predicate = new Equal(numberField, new EqualOp(), 1);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('not.string', () => {
          const predicate = new Equal(stringField, new EqualOp(true), '1');
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('string', () => {
          const predicate = new Equal(stringField, new EqualOp(), '1');
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('not.boolean', () => {
          const predicate = new Equal(booleanField, new EqualOp(true), true);
          expect(FilterBuilder.build(predicate, FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('boolean', () => {
          const predicate = new Equal(booleanField, new EqualOp(), true);
          expect(FilterBuilder.build(predicate, FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('greaterOrEqualThan', () => {
      describe('sql', () => {
        test('number', () => {
          const predicate = new GreaterOrEqualThan(numberField, 1);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('string', () => {
          const predicate = new GreaterOrEqualThan(stringField, '1');
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('greaterThan', () => {
      describe('sql', () => {
        test('number', () => {
          const predicate = new GreaterThan(numberField, 1);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('string', () => {
          const predicate = new GreaterThan(stringField, '1');
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('ilike', () => {
      describe('cql', () => {
        test('not.string', () => {
          const predicate = new Ilike(stringField, new IlikeOp(true), '%bar%');
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });

        test('string', () => {
          const predicate = new Ilike(stringField, new IlikeOp(), '%bar%');
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });
      });
      describe('sql', () => {
        test('not.string', () => {
          const predicate = new Ilike(stringField, new IlikeOp(true), '%bar%');
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('string', () => {
          const predicate = new Ilike(stringField, new IlikeOp(), '%bar%');
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('in', () => {
      describe('cql', () => {
        test('not.number', () => {
          const predicate = new In(numberField, new InOp(true), [1, 2, 3]);
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });
        test('number', () => {
          const predicate = new In(numberField, new InOp(), [1, 2, 3]);
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });

        test('not.string', () => {
          const predicate = new In(stringField, new InOp(true), ['bar1', 'bar2', 'bar3']);
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });
        test('string', () => {
          const predicate = new In(stringField, new InOp(), ['bar1', 'bar2', 'bar3']);
          expect(predicate.toString(FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });
      });
      describe('sql', () => {
        test('not.number', () => {
          const predicate = new In(numberField, new InOp(true), [1, 2, 3]);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
        test('number', () => {
          const predicate = new In(numberField, new InOp(), [1, 2, 3]);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('not.string', () => {
          const predicate = new In(stringField, new InOp(true), ['bar1', 'bar2', 'bar3']);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
        test('string', () => {
          const predicate = new In(stringField, new InOp(), ['bar1', 'bar2', 'bar3']);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('like', () => {
      describe('cql', () => {
        test('not.string', () => {
          const predicate = new Like(stringField, new LikeOp(true), '%bar%');
          expect(FilterBuilder.build(predicate, FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });
        test('string', () => {
          const predicate = new Like(stringField, new LikeOp(), '%bar%');
          expect(FilterBuilder.build(predicate, FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
        });
      });
      describe('sql', () => {
        test('not.string', () => {
          const predicate = new Like(stringField, new LikeOp(true), '%bar%');
          expect(FilterBuilder.build(predicate, FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
        test('string', () => {
          const predicate = new Like(stringField, new LikeOp(), '%bar%');
          expect(FilterBuilder.build(predicate, FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('lowerOrEqualThan', () => {
      describe('sql', () => {
        test('number', () => {
          const predicate = new LowerOrEqualThan(numberField, 1);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('string', () => {
          const predicate = new LowerOrEqualThan(stringField, '1');
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('lowerThan', () => {
      describe('sql', () => {
        test('number', () => {
          const predicate = new LowerThan(numberField, 1);
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('string', () => {
          const predicate = new LowerThan(stringField, '1');
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('null', () => {
      describe('sql', () => {
        test('not', () => {
          const predicate = new Null(numberField, new NullOp(true));
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });

        test('', () => {
          const predicate = new Null(stringField, new NullOp());
          expect(predicate.toString(FilterBuilderTypeEnum.SQL)).toMatchSnapshot();
        });
      });
    });

    describe('or', () => {
      describe('cql', () => {
        // TODO
      });
      describe('sql', () => {
        // TODO
      });
    });
  });

  describe('filterBuilder', () => {
    describe('empty', () => {
      test('', () => {
        expect(FilterBuilder.build(undefined, FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
      });
    });

    describe('multiple', () => {
      describe('cql', () => {
        describe('equal', () => {
          test('number AND string', () => {
            const field1: IField<{ foo: number }> = {
              type: FieldTypeEnum.Number,
              key: 'foo',
            };
            const predicate1 = new Equal(field1, new EqualOp(), 1);
            const field2: IField<{ foo: string }> = {
              type: FieldTypeEnum.String,
              key: 'bar',
            };
            const predicate2 = new Equal(field2, new EqualOp(), 'bar');

            const predicate3 = new AndPre(predicate1, predicate2);

            expect(FilterBuilder.build(predicate3, FilterBuilderTypeEnum.CQL)).toMatchSnapshot();
          });
          test('complex', () => {
            const field1: IField<{ foo: number }> = {
              type: FieldTypeEnum.Number,
              key: 'foo',
            };
            const predicate1 = new Equal(field1, new EqualOp(), 1);

            const field2: IField<{ foo: string }> = {
              type: FieldTypeEnum.String,
              key: 'bar',
            };
            const predicate2 = new Equal(field2, new EqualOp(), 'bar');

            const field3: IField<{ foo: string }> = {
              type: FieldTypeEnum.String,
              key: 'bar3',
            };
            const predicate3 = new Equal(field3, new EqualOp(), 'bar3');

            expect(
              new FilterBuilder(predicate1).and(predicate2).or(predicate3).build(FilterBuilderTypeEnum.CQL)
            ).toMatchSnapshot();
          });
        });
      });
      describe('sql', () => {
        test('complex', () => {
          const dateDebutField: IField<any> = {
            key: 'date_debut',
            type: FieldTypeEnum.String,
          };
          const dateFinField: IField<any> = {
            key: 'date_fin',
            type: FieldTypeEnum.String,
          };
          const predicate1 = new LowerOrEqualThan(dateDebutField, '2020-01-12');
          const predicate2 = new Null(dateFinField, new NullOp());
          const predicate3 = new GreaterThan(dateFinField, '2020-01-12');

          const ids = '1,2,3,4,5';
          const field: IField<any> = {
            key: 'id_dependance',
            type: FieldTypeEnum.String,
          };
          const predicate4 = new In(field, new InOp(), ids);

          expect(
            new FilterBuilder(predicate2)
              .or(predicate3)
              .and(predicate1)
              .and(predicate4)
              .build(FilterBuilderTypeEnum.SQL)
          ).toMatchSnapshot();
        });
      });
    });
  });
});
