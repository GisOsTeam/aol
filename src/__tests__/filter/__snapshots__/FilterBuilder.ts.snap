// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aol.filter filterBuilder empty  1`] = `"INCLUDE"`;

exports[`aol.filter filterBuilder multiple cql equal number AND string 1`] = `"(foo = 1) && (Concatenate(bar) = 'bar')"`;

exports[`aol.filter filterBuilder simple cql equal boolean 1`] = `"(foo = true)"`;

exports[`aol.filter filterBuilder simple cql equal not.boolean 1`] = `"(foo <> true)"`;

exports[`aol.filter filterBuilder simple cql equal not.number 1`] = `"(foo <> 1)"`;

exports[`aol.filter filterBuilder simple cql equal not.string 1`] = `"(Concatenate(foo) <> '1')"`;

exports[`aol.filter filterBuilder simple cql equal number 1`] = `"(foo = 1)"`;

exports[`aol.filter filterBuilder simple cql equal string 1`] = `"(Concatenate(foo) = '1')"`;

exports[`aol.filter filterBuilder simple cql ilike not.string 1`] = `"(Concatenate(foo) NOT ILIKE '%bar%')"`;

exports[`aol.filter filterBuilder simple cql ilike string 1`] = `"(Concatenate(foo) ILIKE '%bar%')"`;

exports[`aol.filter filterBuilder simple cql in not.number 1`] = `"(foo NOT IN (1,2,3))"`;

exports[`aol.filter filterBuilder simple cql in not.string 1`] = `"(foo NOT IN ('bar1','bar2','bar3'))"`;

exports[`aol.filter filterBuilder simple cql in number 1`] = `"(foo IN (1,2,3))"`;

exports[`aol.filter filterBuilder simple cql in string 1`] = `"(foo IN ('bar1','bar2','bar3'))"`;

exports[`aol.filter filterBuilder simple cql like not.string 1`] = `"(Concatenate(foo) NOT LIKE '%bar%')"`;

exports[`aol.filter filterBuilder simple cql like string 1`] = `"(Concatenate(foo) LIKE '%bar%')"`;

exports[`aol.filter filterBuilder simple sql equal boolean 1`] = `"(foo = true)"`;

exports[`aol.filter filterBuilder simple sql equal not.boolean 1`] = `"(foo <> true)"`;

exports[`aol.filter filterBuilder simple sql equal not.number 1`] = `"(foo <> 1)"`;

exports[`aol.filter filterBuilder simple sql equal not.string 1`] = `"(foo <> '1')"`;

exports[`aol.filter filterBuilder simple sql equal number 1`] = `"(foo = 1)"`;

exports[`aol.filter filterBuilder simple sql equal string 1`] = `"(foo = '1')"`;

exports[`aol.filter filterBuilder simple sql ilike not.string 1`] = `"(UPPER(foo) NOT LIKE '%BAR%')"`;

exports[`aol.filter filterBuilder simple sql ilike string 1`] = `"(UPPER(foo) LIKE '%BAR%')"`;

exports[`aol.filter filterBuilder simple sql in not.number 1`] = `"(foo NOT IN (1,2,3))"`;

exports[`aol.filter filterBuilder simple sql in not.string 1`] = `"(foo NOT IN ('bar1','bar2','bar3'))"`;

exports[`aol.filter filterBuilder simple sql in number 1`] = `"(foo IN (1,2,3))"`;

exports[`aol.filter filterBuilder simple sql in string 1`] = `"(foo IN ('bar1','bar2','bar3'))"`;

exports[`aol.filter filterBuilder simple sql like not.string 1`] = `"(foo NOT LIKE '%bar%')"`;

exports[`aol.filter filterBuilder simple sql like string 1`] = `"(foo LIKE '%bar%')"`;
